### ISSUE TITLE
NewIndexAlias to return IndexAlias interface instead

### ISSUE THREAD

**Issue by tugberkugurlu** [blevesearch/bleve#1207]
👋 Is there a reason why https://github.com/blevesearch/bleve/blob/84740560801d65fa8576bbf417ff856f794b884f/index_alias_impl.go#L39 returns the implementation type instead of https://github.com/blevesearch/bleve/blob/84740560801d65fa8576bbf417ff856f794b884f/index_alias.go#L31-L37

**Comment by mschoch** [comment]
I don't think so.  I think we were just trying to follow the general Go guideline of `return structs, accept interfaces`.

https://medium.com/@cep21/what-accept-interfaces-return-structs-means-in-go-2fe879e25ee8

We don't follow that guideline everywhere, or with any real consistency.

**Comment by tugberkugurlu** [comment]
Thanks, it was causing me a problem where I tried to assign `NewIndexAlias` return value to `*bleve.Index` type but I don't believe I need that anymore and I am note sure whether changing it would actually fixed that (also not sure pointers on interfaces make much sense). In any case, I would put my vote on hiding the impl types from the consumer and return public interfaces. I had to look for `bleve.IndexAlias` interface as it wasn't obvious to me there was one.
